api/server/server.go

```diff
@@ -624,6 +624,17 @@ func (s *Server) getContainersLogs(version version.Version, w http.ResponseWrite
 		closeNotifier = notifier.CloseNotify()
 	}
 
+	c, err := s.daemon.Get(vars["name"])
```

Comment:
Yes, the reason I'm doing this is because otherwise we end up returning a 200 even though there is no container.
I think we also did this for the stats endpoint.